### ISSUE TITLE
Do I have to run first RunUMAP or FindClusters?

### ISSUE THREAD

**Issue by MichaelStber** [satijalab/seurat#2152]
Dear all, dear Seurat team,

I have two general questions.

In your vignettes "Integrating stimulated vs. control PBMC" your are running first RunUMAP and then FindNeighbors (setting the reduction to "pca" so that it will not take the UMAP reduction). 
1) For running "FindClusters" you do not specify "reduction = pca" because the clustering is already done with "pca"??? Do I have to add reduction = pca here as well?

`immune.combined <- RunUMAP(immune.combined, reduction = "pca", dims = 1:20)`
`immune.combined <- FindNeighbors(immune.combined, reduction = "pca", dims = 1:20)`
`immune.combined <- FindClusters(immune.combined, resolution = 0.5)`

In the Vignette "Guided Clustering Tutorial" you are running RunUMAP after FindingClusters:
`pbmc <- FindNeighbors(pbmc, dims = 1:10)`
`pbmc <- FindClusters(pbmc, resolution = 0.5)`
`pbmc <- RunUMAP(pbmc, dims = 1:10)`

2) Is that because you are using UMAP here only for visualization? Why is the order different here compared to the other Vignitte?
Just want to understand it =)

Best regards,
Michael

**Comment by andrewwbutler** [comment]
Hi Michael,

1. `FindClusters` performs graph-based clustering on the neighbor graph that is constructed with the `FindNeighbors` function call. This neighbor graph is constructed using PCA space when you specifiy `reduction = "pca"`. You shouldn't add `reduction = "pca"` to `FindClusters`. 

2. Yes, UMAP is used here only for visualization so the order of `RunUMAP` vs `FindClusters` doesn't really matter (you just want to have run both before plotting with `DimPlot` in order to visualize the clusters in UMAP space).

**Comment by singletrips** [comment]
If I try different resolutions with FindClusters(), should I run FindNeighbors() multiple times or I only need to run FindNeighbors() only once?

**Comment by andrewwbutler** [comment]
If you are only changing the `resolution` parameter, you only need to rerun `FindClusters`.

**Comment by erzakiev** [comment]
as we all know, PCA embeds the cells into a N-dimentional representation while respecting the original euclidian distances between the cells, while UMAP contorts the original distances in favor of emphasizing smaller distances. 

We are always told that we should use PCA as a basis for the nearest neighbour calculation, which is more just to the original data, we can all agree. But that approach often leads to situations when cells assigned to the same cluster by FindClusters called on the `pca` embedding end up on the different extremities of the UMAP plot.

What if I wanted to emphasize the local similarity of cells i.e. use UMAP as the basis for my clustering? If UMAP makes sense visually in a 2D/3D space, why should I use it as a basis for my clustering, after all?
